libcst/metadata/tests/test_accessor_provider.py

```diff
@@ -0,0 +1,69 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+#
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+import dataclasses
+
+from textwrap import dedent
+
+import libcst as cst
+from libcst.metadata import MetadataWrapper, AccessorProvider
+from libcst.testing.utils import data_provider, UnitTest
+
+
+class DependentVisitor(cst.CSTVisitor):
+    METADATA_DEPENDENCIES = (AccessorProvider,)
+
+    def __init__(self, *, test: UnitTest) -> None:
+        self.test = test
+
+    def on_visit(self, node: cst.CSTNode) -> bool:
+        for f in dataclasses.fields(node):
+            child = getattr(node, f.name)
+            if type(child) is cst.CSTNode:
+                accessor = self.get_metadata(AccessorProvider, child)
+                self.test.assertEqual(accessor, f.name)
```

Comment:
This is not a very valuable test as it just duplicates the core logic. I suppose that's true for the parent metadata provider tests too

Reply:
> Should I serialise something like a tree of accessors and confirm it gets the right serialization ?

If you have the time, that would be great. No big deal if not.